### ISSUE TITLE
[bug] Duplicate scene ids silently overwrite each other's stored frames

### ISSUE THREAD

**Issue by Rishet11** [robocurve/inspect-robots#289]
**Describe the bug**

Nothing validates that scene ids within a `Task` are unique, but downstream code
derives per-trial identity from them. `rollout.py:247` builds
`trial_id = f"{scene.id}-e{epoch}"`, and `FrameStore.put` turns that into a
filename (`frames.py:59`). Two scenes sharing an id therefore write to the same
frame paths, and the second silently overwrites the first: half the run's stored
frames disappear with no warning, no error, and a log that still claims both
trials ran.

The same assumption is made in `_summarize.py:168-170`, which keys transcripts by
`(scene_id, epoch)`:

```python
transcript_by_key = {
    (transcript.scene_id, transcript.epoch): transcript for transcript in transcripts
}
```

so one duplicate scene's transcript is dropped and the surviving one is reported
twice under both rows.

I'd file this against `Task` rather than either consumer: both consumers are
reasonable to assume unique ids, and patching them individually leaves the next
consumer to rediscover it.

**To reproduce**

A/B, identical except for the scene ids:

```python
import tempfile, os, glob
from inspect_robots import eval as ir_eval
from inspect_robots.task import Task
from inspect_robots.scene import Scene
from inspect_robots.mock import CubePickEmbodiment, ScriptedPolicy
from inspect_robots.scorer import success_at_end

def run(ids, label):
    tmp = tempfile.mkdtemp()
    task = Task(
        name="t",
        scenes=[Scene(id=i, instruction="reach the cube", init_seed=n)
                for n, i in enumerate(ids)],
        scorer=success_at_end(), max_steps=5,
    )
    log = ir_eval(task, policy=ScriptedPolicy(), embodiment=CubePickEmbodiment(),
                  log_dir=tmp, store_frames=True)[0]
    files = sorted(os.path.basename(p)
                   for p in glob.glob(os.path.join(log.stats.frames_dir, "*.npy")))
    prefixes = sorted({f.rsplit('_', 2)[0] for f in files})
    print(f"{label}: scene ids={ids}")
    print(f"   trials={log.results.total_trials}  frame_files={len(files)}  trial_prefixes={prefixes}")
    return len(files)

a = run(["s1", "s2"], "UNIQUE ids  ")
b = run(["same", "same"], "DUPLICATE ids")
print(f"\nframes written with unique ids:    {a}")
print(f"frames written with duplicate ids: {b}")
print(f"SILENTLY LOST: {a - b} frame files")
```

```
UNIQUE ids  : scene ids=['s1', 's2']
   trials=2  frame_files=12  trial_prefixes=['s1-e0', 's2-e0']
DUPLICATE ids: scene ids=['same', 'same']
   trials=2  frame_files=6  trial_prefixes=['same-e0']

frames written with unique ids:    12
frames written with duplicate ids: 6
SILENTLY LOST: 6 frame files
```

The log reports `total_trials: 2` and two samples, both with
`scene_id='same'`, so nothing downstream can tell the two trials apart or notice
that one trial's frames are gone.

**Expected behavior**

`Task` should reject duplicate scene ids at construction with a `ConfigError`
naming the offending id, the way it already rejects every other configuration
that would make a rollout meaningless — `Task.__post_init__` raises `ConfigError`
for a missing/double horizon (`task.py:75`), a `max_steps` that is boolean or
`< 1` (`task.py:79`), a non-finite or non-positive `max_seconds` (`task.py:84`),
and `Epochs.count < 1` (`task.py:32`). Given that level of upfront validation,
silently losing half a run's frames to a duplicate id reads as an oversight
rather than a deliberate permission.

If duplicate ids are meant to be legal, then the frame path and the digest key
both need to incorporate something else that is actually unique (the scene's
index in `task.scenes`), which is a bigger change and would alter existing frame
filenames.

**Environment**
- Inspect Robots version (`inspect-robots --version`): 0.0.1.dev60+g13ca1e41a (main @ 13ca1e4)
- Python version: 3.13.3
- OS: macOS 26.5.1 (arm64)
- NumPy version: 2.4.6
- Optional extras installed (rerun, etc.): `.[dev]` only, no rerun

**Additional context**

Happy to send the fix: a uniqueness check in `Task.__post_init__` raising
`ConfigError` with the duplicate id named, plus a regression test. That is one
guard in one place and leaves both consumers untouched.
